apps/src/applab/dontMarshalApi.js

```diff
@@ -42,10 +83,17 @@ exports.removeItem = function (array, index) {
   if (index < array.length) {
     delete array.properties[array.length - 1];
     array.length -= 1;
+  } else {
+    // index is out of bounds (too large):
+    outputWarning("removeItem() index parameter value (" + index +
+                    ") is larger than the number of items in the list (" +
```

Comment:
Perhaps we should have the same behavior for negative indices which are too large? Currently we delete the first element in that case, but it seems like doing nothing and giving a warning might be more appropriate as it most likely indicates a programming error.